crates/spirv-std/macros/src/lib.rs

```diff
@@ -138,9 +138,16 @@ pub fn Image(item: TokenStream) -> TokenStream {
     output.into()
 }
 
+/// Replaces all (nested) occurrences of the #[spirv(..)] attribute with
+/// #[cfg_attr(target_arch="spirv", rust_gpu::spirv(..))].
 #[proc_macro_attribute]
-pub fn spirv(_attr: TokenStream, item: TokenStream) -> TokenStream {
-    let mut tokens = Vec::new();
+pub fn spirv(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let mut tokens: Vec<TokenTree> = Vec::new();
+
+    // prepend with #[rust_gpu::spirv(..)]
+    let attr: proc_macro2::TokenStream = attr.into();
+    tokens.extend(quote! { #[cfg_attr(target_arch="spirv", rust_gpu::spirv(#attr))] });
```

Comment:
Huh, [this is not what I expected](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) (I think the plan used to be exposing which kind of delimiters were used with the attribute but we gave up on that?).

In that case I suppose the code looks fine? (a bit surprised you need a `proc_macro2::TokenStream`, do the builtin ones not implement the trait `quote!` wants?)

Reply:
Yeah `quote!` only works on `proc_macro2` stuff. Really weird.

Reply:
Wait, what exactly did you not expect?

Reply:
> Wait, what exactly did you not expect?

That from `#[attr_name(inputs)]`, the proc macro gets only `inputs`, not `(inputs)` - my understanding was that we would always include the parens used with the attribute, to allow things like `#[attr_name = "value"]` to receive `= "value"` - I'm likely just confused though.